src/aap_eda/services/ruleset/activate_rulesets.py

```diff
@@ -75,26 +80,37 @@ def find_ports(rulebook_text: str):
 
 
 def save_activation_and_instance(
-    instance: models.ActivationInstance, update_fields: list
+    instance: models.ActivationInstance,
+    update_fields: list,
+    raise_exception: bool = True,
 ):
-    """Save instance and update the linked activation's status accordingly."""
-    instance.activation.status = instance.status
-    running_states = [
-        ActivationStatus.PENDING.value,
-        ActivationStatus.STARTING.value,
-        ActivationStatus.RUNNING.value,
-        ActivationStatus.UNRESPONSIVE.value,
-    ]
-    activation_fields = ["status", "modified_at"]
-    if str(instance.status) not in running_states:
-        instance.activation.current_job_id = None
-        activation_fields.append("current_job_id")
-    if str(instance.status) == ActivationStatus.COMPLETED.value:
-        instance.activation.failure_count = 0
-        activation_fields.append("failure_count")
-
-    instance.save(update_fields=update_fields)
-    instance.activation.save(update_fields=activation_fields)
+    try:
+        """
+        Save instance and update the linked activation's
+        status accordingly.
+        """
+        instance.activation.status = instance.status
+        running_states = [
+            ActivationStatus.PENDING.value,
+            ActivationStatus.STARTING.value,
+            ActivationStatus.RUNNING.value,
+            ActivationStatus.UNRESPONSIVE.value,
+        ]
+        activation_fields = ["status", "modified_at"]
+        if str(instance.status) not in running_states:
+            instance.activation.current_job_id = None
+            activation_fields.append("current_job_id")
+        if str(instance.status) == ActivationStatus.COMPLETED.value:
+            instance.activation.failure_count = 0
```

Comment:
@hsong-rh Shouldn't this be preserved to record the number of times the activation has failed. If in the future we want to display the failed lists of activation we could have a link that shows all the failed activations.

Reply:
We may discuss this in dev sync meeting, the current logic is to reset it to 0 whenever the activation is complete.